network/network_interface.ml

```diff
@@ -25,299 +25,647 @@ let uri () = "file:" ^ !default_path
+  (* Change from ipv4_routes: (Unix.inet_addr * int * Unix.inet_addr) list; *)
+  ipv4_routes: ipv4_route_t list [@default []];
+  dns: Unix.inet_addr list * string list [@default [], []];
+  mtu: int [@default 1500];
+  ethtool_settings: (string * string) list [@default []];
+  ethtool_offload: (string * string) list [@default ["lro", "off"]];
+  persistent_i: bool [@default false];
 }
+[@@deriving rpcty]
+
 type port_config_t = {
-	interfaces: iface list;
-	bond_properties: (string * string) list;
-	bond_mac: string option;
-	kind: port_kind;
-}
+  interfaces: iface list [@default []];
+  bond_properties: (string * string) list [@default []];
+  bond_mac: string option [@default None];
+  kind: port_kind [@default Basic_port];
+} [@@deriving rpcty]
+
 type bridge_config_t = {
-	ports: (port * port_config_t) list;
-	vlan: (bridge * int) option;
-	bridge_mac: string option;
-	igmp_snooping: bool option;
-	other_config: (string * string) list;
-	persistent_b: bool;
-}
-type config_t = {
-	interface_config: (iface * interface_config_t) list;
-	bridge_config: (bridge * bridge_config_t) list;
-	gateway_interface: iface option;
-	dns_interface: iface option;
-}
+  ports: (port * port_config_t) list [@default []];
+  vlan: (bridge * int) option [@default None];
+  bridge_mac: string option [@default None];
+  igmp_snooping: bool option [@default None];
+  other_config: (string * string) list [@default []];
+  persistent_b: bool [@default false];
+} [@@deriving rpcty]
 
-(** {2 Default configuration} *)
+type config_t = {
+  interface_config: (iface * interface_config_t) list [@default []];
+  bridge_config: (bridge * bridge_config_t) list [@default []];
+  gateway_interface: iface option [@default None];
+  dns_interface: iface option [@default None];
+} [@@deriving rpcty]
 
 let default_interface = {
-	ipv4_conf = None4;
-	ipv4_gateway = None;
-	ipv6_conf = None6;
-	ipv6_gateway = None;
-	ipv4_routes = [];
-	dns = [], [];
-	mtu = 1500;
-	ethtool_settings = [];
-	ethtool_offload = ["lro", "off"];
-	persistent_i = false;
+  ipv4_conf = None4;
+  ipv4_gateway = None;
+  ipv6_conf = None6;
+  ipv6_gateway = None;
+  ipv4_routes = [];
+  dns = [], [];
+  mtu = 1500;
+  ethtool_settings = [];
+  ethtool_offload = ["lro", "off"];
+  persistent_i = false;
 }
+
 let default_bridge = {
-	ports = [];
-	vlan = None;
-	bridge_mac = None;
-	igmp_snooping = None;
-	other_config = [];
-	persistent_b = false;
+  ports = [];
+  vlan = None;
+  bridge_mac = None;
+  igmp_snooping = None;
+  other_config = [];
+  persistent_b = false;
 }
+
 let default_port = {
-	interfaces = [];
-	bond_properties = [];
-	bond_mac = None;
-	kind = Basic;
+  interfaces = [];
+  bond_properties = [];
+  bond_mac = None;
+  kind = Basic_port;
 }
+
 let default_config = {
-	interface_config = [];
-	bridge_config = [];
-	gateway_interface = None;
-	dns_interface = None
+  interface_config = [];
+  bridge_config = [];
+  gateway_interface = None;
+  dns_interface = None
 }
```

Comment:
I don't think you need these default_* any more because you've got the in-line defaults defined above.

Reply:
Yes, idl do not need this anymore, but xcp-networkd need them, so I will move them to xcp-networkd repo.